presto-main/src/main/java/com/facebook/presto/operator/aggregation/differentialentropy/StateStrategy.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.differentialentropy;
+
+import io.airlift.slice.SliceOutput;
+
+/**
+ * Interface for different strategies for calculating entropy: MLE (maximum likelihood
+ * estimator) using NumericHistogram, jacknife estimates using a fixed histogram, compressed
+ * counting and Renyi entropy, and so forth.
+ */
+public interface StateStrategy
+        extends Cloneable
+{
+    void add(double sample, double weight);
+
+    double calculateEntropy();
+
+    long estimatedInMemorySize();
+
+    int getRequiredBytesForSerialization();
```

Comment:
I'd remove this method and add a parameter to serialize for state code.

Reply:
Sorry, but I just don't understand the comment. Are you suggesting to add a parameter to a method of `State`?

Reply:
@atavory my suggestion is the following:

- remove `getRequiredBytesForSerialization` method
- add parameter to `serialize` method: `void serialize(int methodCode, SliceOutput output)`
- update implementations of `serialize` to include methodCode in the output

Reply:
@mbasmanova IIUC, we're OK with retaining the current way. If not, do please LMK. Many thanks